all/src/test/java/io/opentelemetry/InternalApiProtectionTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.lang.syntax.elements.ClassesShouldConjunction;
+import org.junit.Test;
+
+public class InternalApiProtectionTest {
+
+  private static final String OTEL_BASE_PACKAGE = "io.opentelemetry";
+  private static final JavaClasses ALL_OTEL_CLASSES =
+      new ClassFileImporter().importPackages(OTEL_BASE_PACKAGE);
+
+  @Test
+  public void contrib_should_not_use_internal_api() {
+    ClassesShouldConjunction contribRule =
+        noClasses()
+            .that()
+            .resideInAPackage("..contrib..")
```

Comment:
I think we should include everything except api/sdk artifacts here.

Reply:
I still did not figured out the correct pattern for that. Can take extra time for this, if needed.

Reply:
Nope. This cannot be done in the scope of this PR. Current module/package structure of Otel is not obvious enough for this.